source/blender/gpu/shaders/gpu_shader_vertex.glsl

```diff
@@ -8,21 +9,24 @@ out block {
 #endif
 
 #ifdef USE_INSTANCING
-in mat3 ininstmatrix;
-in vec3 ininstposition;
-in vec4 ininstcolor;
-in int ininstlayer;
-
-out vec4 varinstcolor;
-out mat4 varinstmat;
-out mat4 varinstinvmat;
-flat out int varinstlayer;
+/* compatibility / in */
```

Comment:
I prefer keep using your nice workaround with macro:
```
#if __VERSION__ < 130
  #define in varying
  #define out varying
#endif
```

But you can move it into gpu_shader_lib.glsl which is included in all shader generate with function GPU_shader_create_ex (so all of them).

The same can apply with workaround of `texelFetch`

Reply:
It not possible to pass a varying to a local function in GLSL or it's a problem with gpu_codegen ?

Reply:
The issue is that the "out" or "in" parameter type in a function is diferent from from in/out or varying type qualifier. 

```
GLSL functions are declared and defined similarly to C/C++ functions. A function declaration in GLSL looks like this:

void MyFunction(in float inputValue, out int outputValue, inout float inAndOutValue);
Functions in GLSL use a calling convention called "value-return." This means that values passed to functions are copied into parameters when the function is called, and outputs are copied out when the function returns.

The in, out, and inout qualifiers are not the same as type qualifiers, even though some of them are named the same. These are parameter qualifiers, and they have a different meaning here.
```